cdap-storage-spi/src/main/java/io/cdap/cdap/spi/data/StructuredTable.java

```diff
@@ -118,14 +118,28 @@ default Collection<StructuredRow> multiRead(Collection<? extends Collection<Fiel
    */
   CloseableIterator<StructuredRow> scan(Range keyRange, int limit) throws InvalidFieldException, IOException;
 
+  /**
+   * Scan a collection of rows from the table matching the partial primary keys.
+   * The rows returned will be sorted on the primary key order.
+   *
+   * @param partialKeys partial primary keys for the scan
+   * @param limit maximum number of rows to return
+   * @return a {@link CloseableIterator} of rows
+   * @throws InvalidFieldException if any of the keys are not part of the table schema, or the types of the value
+   *                               do not match
+   * @throws IOException if there is an error scanning the table
+   */
+  CloseableIterator<StructuredRow> scan(Collection<Field<?>> partialKeys, int limit)
```

Comment:
Do we need this new method? Can the `scan(Range, int)` handle it with a singleton Range from the caller?

Reply:
I thought about it.. however if we directly relax the existing scan Range method, we need to divert inside to check if it's singleton range and then do partial range scan + post filtering. This will contradict the contract that it accepts both singleton range (no prefix restriction) and actual bound ranges (prefix restriction).  Instead, seems like having a well-defined partialKeys scan is clearer

Reply:
Second thought, yeah we could do post filtering using the keys range, working on it now